src/img/load.c

```diff
@@ -0,0 +1,270 @@
+/* DreamShell ##version##
+   pvr.c
+   SWAT
+*/
+
+#include "ds.h"
+#include "SegaPVRImage.h"
+#include "img/load.h"
+#include <kmg/kmg.h>
+#include <zlib/zlib.h>
+
+#if defined(__DREAMCAST__)
+#include <malloc.h>
+#endif
+
+#if defined(HAVE_STDIO_H) && !defined(__DREAMCAST__)
+int fileno(FILE *f);
+#endif
+
+static bool bPVRTwiddleTable = false;
+
+/* Open the pvr texture and send it to VRAM */
+int pvr_to_img(const char *filename, kos_img_t *rv)
+{
+	unsigned char *image;
+	unsigned long int imageSize;
+
+	FILE *pFile = fopen(filename, "rb");
+	if (pFile == NULL)
+	{
+		return -1;
+	}
+
+	fseek(pFile, 0, SEEK_END);			 // seek to end of file
+	size_t fsize = (size_t)ftell(pFile); // get current file pointer
+	fseek(pFile, 0, SEEK_SET);
+
+	uint8 *data = (uint8 *)memalign(32, fsize);
+	if (data == NULL)
+	{
+		fclose(pFile);
+		return -1;
+	}
+
+#if defined(HAVE_STDIO_H) && !defined(__DREAMCAST__)
+
+	int fd = fileno(pFile);
+
+	if (fd > -1)
+	{
+		if (read(fd, data, fsize) < 0)
+		{
+			close(fd);
+			fclose(pFile);
+			return -1;
+		}
+	}
+	else
+	{
+		if (!fread(data, fsize, 1, pFile))
+		{
+			fclose(pFile);
+			return -1;
+		}
+	}
+
+#else
+	if (!fread(data, fsize, 1, pFile))

```

Comment:
Much better use KOS fs instead of stdlib for file operations. Because stdlib reads by small portions and uses cache. You can improve loading speed by replace it to fs_read